framework/doc/content/syntax/HDGKernels/index.md

```diff
@@ -0,0 +1,77 @@
+
+HDG kernels and their boundary condition counterparts,
+[HDGBCs/index.md], are advanced systems that should only be developed by
+users with a fair amount of finite element experience. For background on
+hybridization, we encourage the user to read [!citep](cockburn2009unified) which
+presents a unified framework for considering hybridization of discontinuous
+Galerkin, mixed, and continuous Galerkin methods for elliptic
+problems. [!citep](cockburn2008superconvergent) presents a single-face
+hybridizable discontinuous Galerkin (HDG) method for an elliptic problem, in which a
+non-zero stabilization term is added to only one face of a given
+element. [!citep](nguyen2010hybridizable) presents an HDG method for Stokes
+flow. [!citep](nguyen2011implicit) extends HDG to Navier-Stokes. More HDG
+literature may be found by looking at the research of Bernardo Cockburn, his
+former postdoc Sander Rhebergen, and Rhebergen's former postdoc Tamas
+Horvath. Work by Tan Bui-Thanh on upwind HDG methods, like in
+[!citep](bui2015godunov) is also worth noting.
+
+A hybridized finite element formulation starts with some primal finite element
+discretization. Then some continuity property of the finite element space is
+broken. For instance Raviart-Thomas finite elements may be used to solve a mixed
+formulation description of a Poisson problem. The Raviart-Thomas elements ensure
+continuity of the normal component of the vector field across element faces. We
+break that continuity in the finite element space used in the hybridized method
+and instead introduce degrees of freedom, that live only on the mesh skeleton
+(the faces of the mesh), that are responsible for ensuring the continuity that
+was lost by breaking the finite element space. In libMesh/MOOSE implementation
+terms, when hybridizing the Raviart-Thomas description of the Poisson problem,
+we change from using a `RAVIART_THOMAS` basis to an `L2_RAVIART_THOMAS` basis
+and introduce a `SIDE_HIERARCHIC` variable whose degrees of freedom live on the
+mesh skeleton. We will refer to the variables that exist "before" the
+hybridization as primal variables and the variable(s) that live on the mesh
+skeleton as Lagrange multipliers (LMs) or dual variable(s).
+
+## Implementation in MOOSE
+
+HDG kernels derive from [Kernels](Kernels/index.md). However, the methods
+that must be overridden are quite different. These are `onElement` and
+`onInternalSide`, which implement integrations in the volume of elements and on
+internal faces respectively. External boundary condition integration occurs in
+[HDGBCs/index.md].
+
+Within `onElement` and `onInternalSide`, hybridized kernel developers have eight
+different data structures they need to populate. These are
+
+```
+  /// Matrix data structures for on-diagonal coupling
+  EigenMatrix _PrimalMat, _LMMat;
+  /// Vector data structures
+  EigenVector _PrimalVec, _LMVec;
+  /// Matrix data structures for off-diagonal coupling
+  EigenMatrix _PrimalLM, _LMPrimal;
```

Comment:
Maybe you could mention that some of these are in the `HDGData` class now. I like the design!